Update dependencies and version

mnatsheh opened the pull request:
This is a huge PR where many files were deleted and some were updated to work with the new modules as well as getting rid of rxdb and related modules.

Some follow up work after this:
- Get rid of GraphQL; this should be part of the LW code base and not in manuscripts.
- Update react.
- Remove other LW code from the module and find a better way to pass the data.
- Delete more file; I'm not sure that the stories fit under this repo.

mbartenev-atypon commented on webpack.common.ts:
```diff
@@ -1,77 +0,0 @@
-/*!
```
Why are webpack files are removed completely?

mnatsheh replied:
From what I can tell, the webpack files were being used to build the public website in manuscripts-frontend, which we don't do here.

If I missed something and the files are still needed, let me know.

mbartenev-atypon commented on package.json:
```diff
@@ -12,17 +12,11 @@
-    "build": "concurrently 'yarn:build-dep:*'",
-    "build-dep:cjs": "tsc --outDir dist/cjs --module commonjs --project tsconfig.build.json",
-    "build-dep:es": "tsc --outDir dist/es --declarationDir dist/types --declaration --project tsconfig.build.json",
+    "dev": "npm-run-all build-dep:* --watch",
```
Why going in favour of npm?

mbartenev-atypon commented on src/postgres-data/buildData.ts:
```diff
@@ -245,69 +229,23 @@ const getCollaboratorsData = async (
+  const doc = decoder.createArticleNode()
+  const ancestorDoc = decoder.createArticleNode()
+  const storeData: Partial<state> = {
+    snapshotID: null,
```
That may need a further cleanup later on, to make sure no components are using it and then removing it from the store type.

mbartenev-atypon commented on modernizr.config.json:
```diff
@@ -1,30 +0,0 @@
-{
```
Is it safe to remove this and also .yarnrc and .broweserslistrc?

mnatsheh replied:
I think it's fine to remove the .yarnrc. As for the others, The usage I found for them was when building the website, so I removed them.

If I missed something and the files are still needed, let me know.

IslamJMomani commented on src/hooks/use-comments.ts:
```diff
@@ -1,322 +0,0 @@
-  }))
-}
-
-export const updateComment = (
```
You will need to update the tests ( `/src/hooks/__tests__/use-comments.test.ts`.)

IslamJMomani reviewed (commented):
I see some other files that could be deleted, I will list them here:
- CommentList (`src/components/projects/CommentList.tsx`)
- ProjectPageContainer
- TemplateSelectorModal and all related components from the templates folder (`src/components/templates`) beside (`updateManuscriptTemplate`) In case we are not going to support updating the manuscript template. 

- Collaboration things (`src/components/collaboration`)
- validation file `/src/validation.ts`.

Just a reminder, the config values change needs to be applied in the dashboard too (https://github.com/atypon/leanworkflow-dashboard/blob/master/config/dev.env)

mnatsheh commented:
I was hesitant to remove some unused things, but after talking with Nick I have decided to go through with the removal.

asouqi commented:
can we add `.vscode` folder to .gitignore

asouqi commented:
@mnatsheh can you please rename `@manuscripts/manuscripts-json-schema` to the new one.  and run `yarn lint --fix` there will be still issues I will look at them for the lint.

mnatsheh commented:
> @mnatsheh can you please rename `@manuscripts/manuscripts-json-schema` to the new one. and run `yarn lint --fix` there will be still issues I will look at them for the lint.

I've already renamed it.

mnatsheh commented:
> @mnatsheh are you able to build the project after the recent update to the branch!
> 
> I face this issue with the build:
> 
> ```
> src/components/Inspector.tsx:26:14 - error TS2742: The inferred type of 'InspectorTabs' cannot be named without a reference to '@manuscripts/style-guide/node_modules/@reach/utils'. This is likely not portable. A type annotation is necessary.
> 
> 26 export const InspectorTabs = styled(Tabs)`
> ```

This likely means that you need to `yarn install --prod` in style-guide and body-editor. Make sure no prosemirror or react modules exist under node_modules.

asouqi commented:
> I've already renamed it.

run `yarn typecheck` it's going to return a list of the files with this issue.

mnatsheh commented:
> > I've already renamed it.
> 
> run `yarn typecheck` it's going to return a list of the files with this issue.

Ah it looks like I missed the tests. I have updated them and things seem to be working now.
